src/test/java/uk/gov/hmcts/dts/fact/services/admin/AdminCourtAddressServiceTest.java

```diff
@@ -0,0 +1,153 @@
+                                                                                                                                                  VISIT_OR_CONTACT_US_ADDRESS_TYPE_NAME_CY);
+
+    private static final List<String> TEST_ADDRESS1 = singletonList("1 High Street");
+    private static final List<String> TEST_ADDRESS2 = asList(
+        "High Court",
+        "2 Main Road"
+    );
+
+    private static final List<String> TEST_ADDRESS_CY1 = emptyList();
+    private static final List<String> TEST_ADDRESS_CY2 = emptyList();
+
+    private static final String TEST_TOWN1 = "London";
+    private static final String TEST_TOWN2 = "Manchester";
+    private static final String TEST_POSTCODE1 = "EC1A 1AA";
+    private static final String TEST_POSTCODE2 = "M1 2AA";
+
+    private static final int ADDRESS_COUNT = 2;
+    private static final CourtAddress WRITE_TO_US_ADDRESS = new CourtAddress(WRITE_TO_US_ADDRESS_TYPE_ID, TEST_ADDRESS1, TEST_ADDRESS_CY1, TEST_TOWN1, null, TEST_POSTCODE1);
+    private static final CourtAddress VISIT_US_ADDRESS = new CourtAddress(VISIT_US_ADDRESS_TYPE_ID, TEST_ADDRESS2, TEST_ADDRESS_CY2, TEST_TOWN2, null, TEST_POSTCODE2);
+    private static final List<CourtAddress> EXPECTED_ADDRESSES = asList(WRITE_TO_US_ADDRESS, VISIT_US_ADDRESS);
+
+    private static final Court MOCK_COURT = mock(Court.class);
+    private static final List<uk.gov.hmcts.dts.fact.entity.CourtAddress> COURT_ADDRESSES_ENTITY = asList(
+        new uk.gov.hmcts.dts.fact.entity.CourtAddress(MOCK_COURT, WRITE_TO_US_ADDRESS_TYPE, TEST_ADDRESS1, TEST_ADDRESS_CY1, TEST_TOWN1, null, TEST_POSTCODE1),
+        new uk.gov.hmcts.dts.fact.entity.CourtAddress(MOCK_COURT, VISIT_US_ADDRESS_TYPE, TEST_ADDRESS2, TEST_ADDRESS_CY2, TEST_TOWN2, null, TEST_POSTCODE2));
+
+    private static final String NOT_FOUND = "Not found: ";
+
+    @Autowired
+    private AdminCourtAddressService adminCourtAddressService;
+
+    @MockBean
+    private CourtRepository courtRepository;
+
+    @MockBean
+    private CourtAddressRepository courtAddressRepository;
+
+    @MockBean
+    private AdminAddressTypeService adminAddressTypeService;
+
+    @Test
+    void shouldReturnAllCourtAddresses() {
+        when(MOCK_COURT.getAddresses()).thenReturn(COURT_ADDRESSES_ENTITY);
+        when(courtRepository.findBySlug(COURT_SLUG)).thenReturn(Optional.of(MOCK_COURT));
+
+        final List<CourtAddress> results = adminCourtAddressService.getCourtAddressesBySlug(COURT_SLUG);
+        assertThat(results).hasSize(ADDRESS_COUNT);
+        assertThat(results.get(0)).isEqualTo(VISIT_US_ADDRESS);
+        assertThat(results.get(1)).isEqualTo(WRITE_TO_US_ADDRESS);
+    }
+
+    @Test
+    void shouldSortAllVisitUsAddressesFirst() {
+        final List<uk.gov.hmcts.dts.fact.entity.CourtAddress> courtAddresses = asList(
+            new uk.gov.hmcts.dts.fact.entity.CourtAddress(MOCK_COURT, WRITE_TO_US_ADDRESS_TYPE, emptyList(), emptyList(), null, null, null),
```

Comment:
Does null imply that these parameters are optional, or could use Optionals? Is there a way to reduce the amount of nulls being passed around? Or is this only for the tests?

Reply:
This test is merely testing the sort order based on the address type so it doesn't matter what is the values passed into other fields. I think that is fine. It is for the test only and we don't need to use the other information